climada/util/plot.py

```diff
@@ -290,6 +293,103 @@ def geo_im_from_array(array_sub, coord, var_name, title,
 
     return axes
 
+
+def geo_scatter_categorical(array_sub, geo_coord, var_name, title,
+                            cat_name = None,
+                            pop_name=True, buffer=BUFFER,
+                            extend='neither', proj=ccrs.PlateCarree(),
+                            shapes=True, axes=None, **kwargs):
+    """Map plots for categorical data defined in array(s) over input
+    coordinates. The categories must be a finite set of unique
+    values as can be identified by np.unique() (int, float, strings, ...).
+    
+    The categories are shared among all subplots, i.e. are obtained from 
+    np.unique(array_sub). 
+    Eg.:
+        array_sub = [[1, 2, 1.0, 2], [1, 2, 'a', 'a']] 
+        -> categories mapping is [[0, 2, 1, 2], [0, 2, 3, 3]]
+        
+    Same category: 1 and '1'
+    Different categories: 1 and 1.0
+    
+    
+    Parameters
+    ----------
+    array_sub : np.array(1d or 2d) or list(np.array)
+        Each array (in a row or in  the list) are categorical values at
+        each point in corresponding geo_coord that are ploted in one subplot.
+    geo_coord : 2d np.array)
+        (lat, lon) for each point in a row. 
+        The same grid is used for all subplots.
+    var_name : str or list(str)
+        label to be shown in the colorbar. If one provided, the same is 
+        used for all subplots. Otherwise provide as many as subplots in
+        array_sub.
+    title : str or list(str)
+        Subplot title. If one provided, the same is
+        used for all subplots. Otherwise provide as many as subplots in
+        array_sub.
+    cat_name : dict
+        Categories name for the colorbar labels. 
+        Keys are all the unique values in array_cub, values are their labels.
+    pop_name  : bool, optional
+        add names of the populated places. The default is True.
+    buffer  : float, optional
+        border to add to coordinates. The default is False.
+    extend : str, optional
+        extend border colorbar with arrows. The default is 'neither'.
+        options: [ 'neither' | 'both' | 'min' | 'max' ]
+    proj : ccrs
+        coordinate reference system used in coordinates
+    shapes : bool, optional
+        Overlay Earth's countries coastlines to matplotlib.pyplot axis.
+        The default is True.
+    kwargs : optional
+        arguments for hexbin matplotlib function.
+
+    Returns
+    -------
+        cartopy.mpl.geoaxes.GeoAxesSubplot
+        
+    """
+    
+    #default cmap
+    cmap_name = 'Accent'
```

Comment:
Accent isn't a particular good choice if you don't happen to have exactly 8 categories, right? And it's really bad if you have more than 8 categories. Maybe the most generally useful solution would be to manually assemble a colormap containing the right number of colors.

Reply:
Here is how you would define a colormap from a list of colors (from https://matplotlib.org/stable/tutorials/colors/colormap-manipulation.html#creating-listed-colormaps):
```
cmap = ListedColormap(["darkorange", "gold", "lawngreen", "lightseagreen"])
```

Reply:
Yes, you are right. However I prefer not to make colormaps myself, as this is not such an easy thing to do correctly (color blindness, black-white shades, good differentiation etc...). I will see whether I can make a good one using colorbrew.

Reply:
Yes, of course, you shouldn't reinvent the wheel but take the first `N` colors from an existing colormap.

Reply:
Imho, it would be good to include `seaborn` into climada for the plotting.

Reply:
For the time being, I replaced the default with 'set3' which has 12 colors.

Reply:
I now use 'brg' which has 256 distinct colors. That should do it for the moment. If you have another suggestion for a super good set of colors (maybe @zeliest you can have a look whether some standard color palette could be defined for CLIMADA?) please let me know.